tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/negative/InvalidConfigurableCliArgsTestCase.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2023, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.testerina.test.negative;
+
+import org.ballerinalang.test.context.BMainInstance;
+import org.ballerinalang.test.context.BallerinaTestException;
+import org.ballerinalang.test.context.LogLeecher;
+import org.ballerinalang.testerina.test.BaseTestCase;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.ballerinalang.test.context.LogLeecher.LeecherType.ERROR;
+
+/**
+ * Test class containing negative test cases related to configurable cli arguments.
+ */
+
+public class InvalidConfigurableCliArgsTestCase extends BaseTestCase {
+    private BMainInstance bMainInstance;
+    private String testFileLocation;
+
+    @BeforeClass
+    public void setup() throws BallerinaTestException {
+        bMainInstance = new BMainInstance(balServer);
+        testFileLocation = projectBasedTestsPath.resolve("configurable-cli-args-test").toString();
+    }
+
+    @Test
+    public void invalidConfigurableCliArgsTest() throws BallerinaTestException {
+        LogLeecher errorLeecher1 = new LogLeecher("[intVar=test] configurable variable 'intVar' is " +
+                "expected to be of type 'int', but found 'test'", ERROR);
```

Comment:
Is this error thrown by the runtime APIs?

Reply:
Yes, the errors are thrown at runtime